File: hex/hex.go
package main

import "os"
import "fmt"
import "strconv"

func main() {
	argsWOProg := os.Args[1:]

	if len(argsWOProg) == 0 {
		fmt.Println()
		fmt.Println("No input")
		fmt.Println()
		os.Exit(0)
	}

	maxLen := 0

	fmt.Println()
	for _, x := range argsWOProg {
		if len(x) > maxLen {
			maxLen = len(x)
		}
	}

	for _, x := range argsWOProg {
		val, err := strconv.ParseUint(x, 10, 64)
		fmt.Printf("%-*s => ", maxLen, x)
		if err != nil {
			fmt.Println(err)
		} else {
			fmt.Printf("%s%s", "0x", strconv.FormatUint(val, 16))
			fmt.Println()
		}
	}
	fmt.Println()
}

File: hex/hex.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <sys/stat.h>
#include <unistd.h>
#include <errno.h>
#include <sysexits.h>
#include <stdbool.h>

#ifdef INPUT_HEX
#define INPUT_BASE 16
#else
#define INPUT_BASE 10
#endif

const char * find_str_end(const char *str){
    while (*str) str++;
    return str;
}

int main(int argc, const char * argv[])
{
    int i;
    int maxLen = 0;
    if (argc == 1) {
        printf("No input\n");
        return 0;
    }

    for (i = 1; i < argc; ++i) {
        if (strlen(argv[i]) > maxLen) {
            maxLen = strlen(argv[i]);
        }
    }

    putchar('\n');
    for (i = 1; i < argc; ++i) {
        const char * endptr = find_str_end(argv[i]);
        char * afterNum;
        uint64_t val = strtoll(argv[i], &afterNum, INPUT_BASE);
        printf("%-*s => ", maxLen, argv[i]);
        if (endptr != afterNum) {
            printf("PARSE ERROR\n");
        }
        else if ((val == 0) && (errno)) {
            printf("RANGE ERROR\n");
            errno = 0;
        }
        else {
#ifdef INPUT_HEX
            printf("%*lld\n", maxLen+1, val);
#else
            printf("%#*llx\n", maxLen+1, val);
#endif

        }
    }
    putchar('\n');

    return 0;
}
